### PR TITLE
CodyGarciaa/Prea-Pages-Cont

### PR DESCRIPTION
**Pull request by CodyGarciaa**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- the pages that show up on the PreaPages screen are now reflective of what is in our supabase
- the posters are still the same, but the names and data connected to each page is based on the supabase
- also, the spanish and english tabs work now, only having every page under its relative language according to the supabase
- also, the pages are put in order before rendered on the screen

- for the video page side, the video plays in relation to which page is pressed
- the next and prev buttons now work for the next page in the order of pages in the relative language

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

IMG_0653
IMG_0652

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- review LegalRights's index.tsx, and then VideoPage's index.tsx
- had to change things up in LegalRights first, and then work to pass data to VideoPage from LegalRights onPress
- so sorry there's no commits because I was accidentally coding/committing on my remote main rip
- feed back on: does all the videos play all the time? when I was testing, it's like 60% of the time playing on my end

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
- erm maybe conflict with Charlotte's navigation, lowk hella hardcoded my navigation in app.tsx bc u said so :)

CC: @philipye314

### CODE REVIEW

File: src/screens/LegalRights/VideoPage/index.tsx [hunk 1]
```diff
@@ -1,12 +1,69 @@
...
+      id: 'string',
+      is_short_answer: true,
+      page_number: 0,
+      parent_id: 'string',
+      short_answer: 'string',
+      spanish: false,
+      survey: 'string',
+      video_id: 'Section Title 1',
+    },
+  ]);
```
Comment (philipye314) [inline src/screens/LegalRights/VideoPage/index.tsx]: try to think about what data you *actually* need in this page, also look into `Typescript` interfaces so we don't use `any` too often

File: src/screens/LegalRights/index.tsx [hunk 2]
```diff
@@ -1,114 +1,83 @@
...
+    {
+      id: 'string',
+      is_short_answer: true,
+      page_number: 0,
+      parent_id: 'string',
+      short_answer: 'string',
+      spanish: false,
+      survey: 'string',
+      video_id: 'Section Title 1',
+    },
```
Comment (philipye314) [inline src/screens/LegalRights/index.tsx]: same thing here

File: src/screens/LegalRights/VideoPage/index.tsx [hunk 3]
```diff
@@ -1,12 +1,69 @@
...
+  // read in all the variables passed in from legal rights page; only once when initally rendered
+  useEffect(() => {
+    setPreaData(testProp['route']['params'][0]);
+    setIndex(testProp['route']['params'][1]);
+    setLanguage(testProp['route']['params'][2]);
+  }, []);
+
+  useEffect(() => {
+    const fetchVideoLink = async () => {
+      let response = supabase.storage
```
Comment (philipye314) [inline src/screens/LegalRights/VideoPage/index.tsx]: in general, supabase queries should be abstracted away from the screens. Write this function in `/supabase/generalQueries`, or, it might be a good idea to make a dedicated file for storage queries as well

File: src/screens/LegalRights/index.tsx [hunk 4]
```diff
@@ -1,114 +1,83 @@
...
+    }
+    const newEnglishModules = englishData;
+    englishData.sort((a, b) => a.page_number - b.page_number); // sort the array based on pages' page_number yur
+
+    const spanishResponse = await supabase // grab all the spanish pages from the supabase table yk
+      .from('prea_page')
+      .select()
+      .eq('spanish', true);
+    const { data: spanishData, error: spanishError } = spanishResponse;
+    if (spanishError) {
```
Comment (philipye314) [inline src/screens/LegalRights/index.tsx]: same supabase comment!

**Review by philipye314 — COMMENTED** [review]
looks good, just some minor changes to fix

**Review by philipye314 — COMMENTED** [review]
I started doing work on some stuff (mainly pulling main into your branch), so be sure to try and match all the different typing now that we have nav. Let me/Charlotte know if you have any questions?

**Review by philipye314 — DISMISSED** [review]
looks good. We'll talk a little bit about styling + supabase changes in the next sprint but this is very good work overall!
